### PR TITLE
Add deterministic SkAllEstimator test

### PR DESCRIPTION
**Pull request by jptowns**
Please review the [developer documentation](https://github.com/QMCPACK/qmcpack/wiki/Development-workflow)
on the wiki of this project that contains help and requirements.

## Proposed changes

Added a deterministic unit test for SkAll in QMCHamiltonians. Initial step towards addressing issue #985.

## What type(s) of changes does this code introduce?

Added a new file "test_skallestimator.cpp" in src/QMCHamiltonians/tests
Modified file "CMakeLists.txt" in src/QMCHamiltonians/tests
Modified file "SkAllEstimator.cpp" and "SkAllEstimator.h" in src/QMCHamiltonians

### Does this introduce a breaking change?

No.

## What systems has this change been tested on?

Tested on x86_64 ubuntu 18LTS workstation with gcc9.2 and the following spack modules:
Currently Loaded Modulefiles:
 1) cmake-3.16.1-gcc-9.2.1-ollsozd    5) netlib-lapack-3.8.0-gcc-9.2.1-zhymqjo  
 2) boost-1.70.0-gcc-9.2.1-rlvtv3m    6) fftw-3.3.8-gcc-9.2.1-3yeowd6           
 3) openmpi-3.1.4-gcc-9.2.1-o4jcilz   7) hdf5-1.10.5-gcc-9.2.1-ppfocqf          
 4) zlib-1.2.11-gcc-9.2.1-rreqoav     8) libxml2-2.9.9-gcc-9.2.1-ezjk7gg        

New test passes all variants of real/complex aos/soa fp/mp.

## Checklist

- Yes This PR is up to date with current the current state of 'develop'
- Yes Code added or changed in the PR has been clang-formatted
- Yes This PR adds tests to cover any new code, or to catch a bug that is being fixed
- No Documentation has been added (if appropriate)

### CODE REVIEW

File: src/QMCHamiltonians/SkAllEstimator.cpp [hunk 1]
```diff
@@ -105,7 +104,7 @@ void SkAllEstimator::evaluateIonIon()
 
 SkAllEstimator::Return_t SkAllEstimator::evaluate(ParticleSet& P)
 {
-  RealType w = tWalker->Weight;
+  RealType w = tWalker->Weight; // Segfaults unless setHistories() is called prior
```
Comment (ye-luo) [inline src/QMCHamiltonians/SkAllEstimator.cpp:107]: Could you move the comment above this line?

File: src/QMCHamiltonians/SkAllEstimator.cpp [hunk 2]
```diff
@@ -145,12 +144,12 @@ SkAllEstimator::Return_t SkAllEstimator::evaluate(ParticleSet& P)
       values[NumK + 2 * k + 1] = w * RhokTot_i[k];
     }
   }
-#else
+#else // Test case only touches this path at the moment
```
Comment (ye-luo) [inline src/QMCHamiltonians/SkAllEstimator.cpp:147]: We always use macro USE_REAL_STRUCT_FACTOR. Why the test case touch #else?

File: src/QMCHamiltonians/SkAllEstimator.cpp [hunk 3]
```diff
@@ -145,12 +144,12 @@ SkAllEstimator::Return_t SkAllEstimator::evaluate(ParticleSet& P)
...
   //sum over species
   std::copy(P.SK->rhok[0], P.SK->rhok[0] + NumK, RhokTot.begin());
   for (int i = 1; i < NumeSpecies; ++i)
     accumulate_elements(P.SK->rhok[i], P.SK->rhok[i] + NumK, RhokTot.begin());
-  Vector<ComplexType>::const_iterator iit(RhokTot.begin()), iit_end(RhokTot.end());
+  //Vector<ComplexType>::const_iterator iit(RhokTot.begin()), iit_end(RhokTot.end());
```
Comment (ye-luo) [inline src/QMCHamiltonians/SkAllEstimator.cpp:152]: If not needed, remove it please.

File: src/QMCHamiltonians/tests/CMakeLists.txt [hunk 4]
```diff
@@ -23,9 +23,20 @@ EXECUTE_PROCESS(COMMAND ${CMAKE_COMMAND} -E make_directory "${UTEST_DIR}")
 MAYBE_SYMLINK(${UTEST_HDF_INPUT} ${UTEST_DIR}/pwscf.pwscf.h5)
 
 
-SET(SRCS test_bare_kinetic.cpp test_coulomb_pbcAB.cpp test_coulomb_pbcAB_ewald.cpp test_coulomb_pbcAA.cpp
-         test_coulomb_pbcAA_ewald.cpp test_force.cpp test_force_ewald.cpp test_ecp.cpp
-         test_hamiltonian_pool.cpp test_hamiltonian_factory.cpp test_PairCorrEstimator.cpp)
+SET(SRCS test_bare_kinetic.cpp
+	 test_coulomb_pbcAB.cpp
```
Comment (ye-luo) [inline src/QMCHamiltonians/tests/CMakeLists.txt:27]: There are tabs in this file.

File: src/QMCHamiltonians/SkAllEstimator.cpp (hunk 1 above)
Comment (jptowns) [reply src/QMCHamiltonians/SkAllEstimator.cpp]: Fixed.

File: src/QMCHamiltonians/SkAllEstimator.cpp (hunk 2 above)
Comment (jptowns) [reply src/QMCHamiltonians/SkAllEstimator.cpp]: Fixed.

File: src/QMCHamiltonians/SkAllEstimator.cpp (hunk 3 above)
Comment (jptowns) [reply src/QMCHamiltonians/SkAllEstimator.cpp]: Removed.

File: src/QMCHamiltonians/tests/CMakeLists.txt (hunk 4 above)
Comment (jptowns) [reply src/QMCHamiltonians/tests/CMakeLists.txt:27]: Oops. Forgot to run clang-format on this file. Fixed.

File: src/QMCHamiltonians/CMakeLists.txt [hunk 5]
```diff
@@ -15,101 +15,48 @@
 #//
```
Comment (ye-luo) [inline src/QMCHamiltonians/CMakeLists.txt:15]: @jptowns please revert this file. It is largely broken.

File: src/QMCHamiltonians/tests/CMakeLists.txt (hunk 4 above)
Comment (jptowns) [reply src/QMCHamiltonians/tests/CMakeLists.txt]: Note to self: do not run clang-format on CMakeLists.txt

**Pull request merged**